### PR TITLE
Feat/show products

### PR DESCRIPTION
**Pull request by devBK84**

### CODE REVIEW

**Comment by MychajloBear** [comment]
I think merging is blocked, sonarcloud issue... cannot accept it

File: backend/src/main/java/com/github/devbk84/backend/models/Order.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.github.devbk84.backend.models;
+
+import java.util.List;
+
+public record Order(
+        String id,
+        String payment,
+        List<Product> products,
+        String ordertBy
```
Comment (s-gbz) [inline backend/src/main/java/com/github/devbk84/backend/models/Order.java:9]: This is supposed to reference the customer who placed the Order I guess?
Perhaps you could think if there's another name for this attribute

Comment (s-gbz) [reply backend/src/main/java/com/github/devbk84/backend/models/Order.java:9]: (see if you could find a name that's a little more precise)

File: backend/src/main/java/com/github/devbk84/backend/models/Order.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.github.devbk84.backend.models;
+
+import java.util.List;
+
+public record Order(
```
Comment (s-gbz) [inline backend/src/main/java/com/github/devbk84/backend/models/Order.java:5]: Don't forget the @Document notation if you aim to have an Orders collection

File: backend/src/main/java/com/github/devbk84/backend/models/Product.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.math.BigDecimal;
+
+@Document("products")
+public record Product(
+        @Id
+        String name,
+        String id,
+        String description,
+        String orderfavorites,
```
Comment (s-gbz) [inline backend/src/main/java/com/github/devbk84/backend/models/Product.java:14]: What is this supposed to describe?

File: backend/src/test/java/com/github/devbk84/backend/controller/ShopControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    private MockMvc mockMvc;
+
+    @Autowired
+    private WebApplicationContext context;
+
+    @Autowired
+    private ShopRepo shopRepo;
+
+    @Autowired
+    private ObjectMapper objectMapper;
```
Comment (s-gbz) [inline backend/src/test/java/com/github/devbk84/backend/controller/ShopControllerTest.java:30]: You don't need any of these at the moment

File: backend/src/test/java/com/github/devbk84/backend/controller/ShopControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+
+
+    @Test
+    @DirtiesContext
+    void getAllProducts() throws Exception {
+        mockMvc.perform(get("/api/products"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[]"));
+    }
```
Comment (s-gbz) [inline backend/src/test/java/com/github/devbk84/backend/controller/ShopControllerTest.java:40]: You might want to describe the expected scenario in the test's name.
Good use of new lines though :+1: Gives a good structure

**Review by s-gbz — COMMENTED** [review]
You've set up the basic structure - good job Sam!
Take a closer look at my comments and feel free to ask for a chat if there's something unclear

**Pull request merged**